### ISSUE TITLE
AccountNotFounds potentially caused by snapshots

### ISSUE THREAD

**Issue by carllin** [solana-labs/solana#5069]
#### Problem
We see AccountNotFounds on the TDS testnet and our own testnets. Last known PR to cause . this was https://github.com/solana-labs/solana/pull/4797

**Comment by mvines** [comment]
We might need to fix this on 0.16 for TdS

**Comment by sambley** [comment]
@carllin, I have not been able to reproduce the issue yet. Could you provide me with the exact steps that you had used to reproduce the issue

**Comment by carllin** [comment]
Yeah, I was running this commit: 44a572416d94032109c585403baa3291ad28ab21, and the testnet setup:

./gce.sh create -n 5 -c 2 -G "--machine-type n1-standard-16 --accelerator count=2,type=nvidia-tesla-v100" -p testnet-dev-carl -d pd-ssd

 ./net.sh restart -f "cuda" -o noValidatorSanity -c "bench-tps=1=--tx_count 200000"

**Comment by mvines** [comment]
I can repro this on the TdS testnet, but with snapshots are disabled.

**Comment by mvines** [comment]
Here's an example ANF I've seen from ^, it's a simple Transfer instruction

```
[2019-07-16T20:18:27.582430975Z WARN  solana::blocktree_processor] Unexpected validator error: AccountNotFound, tx: Transaction { sign
atures: [5KAUDYsSvJ2Hi141g5ngKqEfTRiFfQvL1cUAHP6FnLXHYAS2m2NrfTjeJskUC9REG1sDBsWLWhVcTd6rJhJ6ZdC6], message: Message { header: Message
Header { num_required_signatures: 1, num_credit_only_signed_accounts: 0, num_credit_only_unsigned_accounts: 1 }, account_keys: [DEqf5GfntbY66eAVEgmUELx1CXVdVtCXjsJTUbJpHbcB, BnqbN8upSZr1r62BaA65n9YHGJYoGwCyg3pyHgniGvhG, 11111111111111111111111111111111], recent_blockhash: HQwbocjqp1w6b3WucW7Pp4eRQciN44YMiRn7VN2LPKRa, instructions: [CompiledInstruction { program_ids_index: 2, accounts: [0, 1], data: [0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] }] } }
```

It does have a credit-only account in it, which after snapshots is the our next favorite punching bag :)

**Comment by CriesofCarrots** [comment]
Actually, this is a `SystemInstruction::CreateAccount` instruction (`data[0..3] = [0, 0, 0, 0]`), which doesn't use credit-only for the To account. The program_id is technically counted among the credit-only accounts for the purposes of the message header, but handled separately as an executable. So I don't think credit-only is to blame... this time...

**Comment by mvines** [comment]
Sorta resolved by disabling snapshots on v0.16

@carllin - is it helpful to keep this open for v0.17?
